internal/web/context.go

```diff
@@ -0,0 +1,22 @@
+package web
+
+import (
+	"github.com/labstack/echo/v4"
+	"go.uber.org/zap"
+)
+
+type AppContext struct {
+	echo.Context
+	AppLogger *zap.Logger
+}
+
+func CreateAppContext(
+	logger *zap.Logger,
+) echo.MiddlewareFunc {
```

Comment:
The CreateAppContext function lacks error handling for the logger argument. Ensure the logger is not nil before proceeding, to prevent a potential nil pointer dereference later in the code.